authentik/policies/unique_password/tasks.py

```diff
@@ -0,0 +1,68 @@
+from structlog import get_logger
+
+from authentik.events.system_tasks import SystemTask, TaskStatus, prefill_task
+from authentik.policies.models import PolicyBinding
+from authentik.policies.unique_password.models import UniquePasswordPolicy, UserPasswordHistory
+from authentik.root.celery import CELERY_APP
+
+LOGGER = get_logger()
+
+
+@CELERY_APP.task(bind=True, base=SystemTask)
+@prefill_task
+def purge_password_history_table(self: SystemTask):
+    """Remove all entries from the UserPasswordHistory table"""
+    unique_pwd_policy_bindings = PolicyBinding.in_use.for_policy(UniquePasswordPolicy)
+
+    if unique_pwd_policy_bindings.count() > 1:
+        # No-op; A UniquePasswordPolicy binding other than the one being deleted still exists
+        self.set_status(
+            TaskStatus.SUCCESSFUL,
+            """Did not purge UserPasswordHistory table. 
+            Bindings for Unique Password Policy still exist.""",
+        )
+        return
+
+    try:
+        # n.b. a performance optimization to execute TRUNCATE
+        # instead of all().delete() would eliminate any FK checks.
+        UserPasswordHistory.objects.all().delete()
+    except Exception as err:
+        LOGGER.debug("Failed to purge UserPasswordHistory table.")
+        self.set_error(err)
+        return
```

Comment:
```suggestion
    UserPasswordHistory.objects.all().delete()
```

Tasks are not auto-retried unless specified. It's fine to not retry this, it doesn't really matter if there are some unused hashes until the next run, especially if this task runs on a schedule.